Mathlib/Topology/Constructions.lean

```diff
@@ -635,6 +635,27 @@ theorem ContinuousAt.prod_map' {f : X → Z} {g : Y → W} {x : X} {y : Y} (hf :
   hf.fst'.prod hg.snd'
 #align continuous_at.prod_map' ContinuousAt.prod_map'
 
+theorem ContinuousAt.comp₂ {f : Y × Z → W} {g : X → Y} {h : X → Z} {x : X}
+    (hf : ContinuousAt f (g x, h x)) (hg : ContinuousAt g x) (hh : ContinuousAt h x) :
+    ContinuousAt (fun x ↦ f (g x, h x)) x :=
+  ContinuousAt.comp hf (hg.prod hh)
+
+theorem ContinuousAt.comp₂_of_eq {f : Y × Z → W} {g : X → Y} {h : X → Z} {x : X} {y : Y × Z}
+    (hf : ContinuousAt f y) (hg : ContinuousAt g x) (hh : ContinuousAt h x) (e : (g x, h x) = y) :
+    ContinuousAt (fun x ↦ f (g x, h x)) x := by
+  rw [← e] at hf
+  exact hf.comp₂ hg hh
+
+/-- Continuous functions on products are continuous in their first argument -/
+theorem Continuous.along_fst {f : X × Y → Z} (hf : Continuous f) {y : Y} :
+    Continuous fun x ↦ f (x, y) :=
+  hf.comp (continuous_id.prod_mk continuous_const)
+
+/-- Continuous functions on products are continuous in their second argument -/
+theorem Continuous.along_snd {f : X × Y → Z} (hf : Continuous f) {x : X} :
```

Comment:
Hmm, actually `continuous_curry_left` and `continuous_curry_right` both exist, but are poorly named (since they're not `Continuous.<something>`.  Should I rename them to `Continuous.curry_left` and `Continuous.curry_right`?

Also, a while ago I added [`AnalyticOn.along_fst`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Analysis/Analytic/Constructions.html#AnalyticOn.along_fst) and similar.  If desired I can add aliases called `AnalyticOn.along_fst` in a separate CL.

Reply:
I can't find `continuous_curry_left` nor `continuous_curry_right`. There are `continuous_uncurry_left` and `continuous_uncurry_right`, but it's something slightly different. In any case, I agree that all these lemmas are poorly named, and that renaming them (and deprecating the previous names) is definitely a good idea!

Reply:
Fixed!